### DISCUSSION TITLE
What does fsck actually check?

### DISCUSSION THREAD

**Discussion by hstock** [juicedata/juicefs#5265] [Q&A]
Hi,

AFAIK fsck checks the metadata but it also accesses s3 storage. On S3, does it check only the presence of an object or does it also verify size and/or s3 provided checksums / etag?

**Comment by jiefenghuang** [comment]
for now, fsck only check the presence of an object.

**Comment by bjo81** [comment]
Und what does repair do then? Throw the object from the metadata?

**Reply by jiefenghuang** [reply]
Adding the repair parameter can fix issues caused by historical problems with dir stat (asynchronous updates) and nlink.
